### DISCUSSION TITLE
meu Dotnet run não funciona direito

### DISCUSSION THREAD

**Discussion by JoaoLCezar** [balta-io/forum#187] [Tira Dúvidas] [answered]
Meu dotnet run nao funciona direito, eu tento executar ele no terminal e é como se ele nao atualizasse meu codigo, fica apenas mostrando hello word ate eu começar o debugging ou apenar F5, com isso ele meio que atualiza o codigo e quando eu dou dotnet run no terminal ele mostra o que esta no codigo, so que as proximas informaçoes não vão sozinhas, eu toda hora que eu mexer no codigo tenho que ficar dando F5 para meu dotnet run funcionar, se for possivel me ajudar, estou um tempo com esse problema, ele chega a ser meio irritante, porque gasta muito tempo que poderia estar estudando ou escrevendo mais codigos, desde ja agradeço.

**Comment by andrebaltieri** [comment, marked as answer]
Nom dia, @JoaoDeLima22, poderia por favor adicionar alguns prints para ajudar a esclarecer o problema?

**Comment by JoaoLCezar** [comment]
![Captura de tela 2022-12-28 224530](https://user-images.githubusercontent.com/118327719/209892807-6e3aa063-7da7-44a7-a656-f50499e07aea.png)
aqui tem o print de um novo projeto que eu mudei uma coisa simple e realizei um dotnet run, a mensagem que deveria mudar nao muda, para isso mudar eu tenho que ficar dando F5 em todo momento ao invez de usar o dotnet run, porem isso faz o prejeto render bem menos

**Reply by andrebaltieri** [reply]
Bom dia, @JoaoDeLima22, o `dotnet run` apenas roda o projeto, o `dotnet watch run` fica observando as mudanças.
Então para obter o comportamento que deseja, precisa rodar o comando `dotnet watch run`.

**Comment by JoaoLCezar** [comment]
Boa Noite, Mesmo assim nao teve o resultado esperado, eu gostaria que quando eu fizesse alteração no meu codigo e testasse com o dotnet run ele ja atualizasse e rodasse o codigo que foi escrito, pois atualmente eu tenho que ficar tando F5 para ele meio que "atualizar" o codigo para poder dar um dotnet run com o codigo com as alterações e isso leva um tempo a mais que eu poderia estar estudando ou fazendo mais linhas, alem de que ficar dendo F5 esta me causando mais erros presentes no codigo. Caso precise de mais prints ou um video mostrando posso tentar enviar.

**Reply by andrebaltieri** [reply]
Bom dia, @JoaoDeLima22, conforme comentei, quem faz este comportamento que você está querendo é o `dotnet watch run`. Foi este comando que você executou?

**Reply by JoaoLCezar** [reply]
Bom dia, sim eu utilizei ele e não houve o resultado procurado, o que eu estou procurando é apenas que faça igual nos vídeos, nos vídeos você escreve o código e da apenas um dotnet run e ele já executa o código, entretanto no meu eu tenho que ficar fazendo o debug se não me engano, que é apertar o F5, aí ele meio que carrega o código e com isso eu posso usar o dotnet run, porém eu tenho que fazer isso toda vez que faço alteração no código.

**Reply by andrebaltieri** [reply]
Certo @JoaoDeLima22, no modo **Debug** é assim mesmo, não tem como alterar e ele já refletir no site, é preciso compilar novamente a aplicação. Por isto, muitas vezes eu mostro com o `dotnet watch run`, por ser mais prático.

**Reply by JoaoLCezar** [reply]
Sim, utilizei ele e mesmo assim não teve o resultado esperado

**Reply by andrebaltieri** [reply]
Bom dia, @JoaoDeLima22, como vai?
Você consegue gravar a tela e enviar aqui? Talvez tenha algo parado, algum erro no VS Code...

Grava o processo:
- Executar o comando `dotnet watch run`
- Abrir o Browser
- Alterar o código
- Volta no terminal para ver se ele reiniciou o App
- Volta no Browser para ver se as alterações foram aplicadas

**Reply by marcelloalonso** [reply]
@JoaoDeLima22 João, boa tarde!! td bem? Conseguiu resolver esse problema?

**Comment by marcelloalonso** [comment]
@andrebaltieri Boa noite! primeiramente agradecer pelo conteúdo!! Estou nos fundamentos do C#. 

Acontece que não consigo tb ver as mudanças que realizo no código através do console do VSCode. 
Por mais que eu mude o Console.WriteLine("Para qualquer texto"); continua exibindo "Hello, World!" no terminal do VSCode, sendo que nem existe esse comando mais na minha aplicação. Já fiz dotnet clean e outros comandos e continua dessa forma.

Não está exibindo o comando atualizado, nem mesmo atribuindo o texto a uma variável: 
![image](https://github.com/balta-io/forum/assets/135472471/017b6072-3d03-4741-a8dc-5d158987e573)

Em todas as aulas os seus comandos são instantaneamente vistos no console após o run.. o que estou errando? 
Obrigado

**Reply by marcelloalonso** [reply]
Mesmo no WindowsTerminal, PowerShell

![image](https://github.com/balta-io/forum/assets/135472471/bd9532f2-51b9-478d-9f55-c5d2c806237a)

**Reply by marcelloalonso** [reply]
Eu preciso dar um "RUN AND DEBUG" toda vez que mudo algo no código, se quiser ver o resultado imediatamente depois através do "dotnet run". Sem isso o resultado em tela nao muda.

**Reply by andrebaltieri** [reply]
@marcelloalonso o comando "dotnet watch run" faz isto, porém ele só funciona com .NET 6 ou superior.

**Reply by marcelloalonso** [reply]
Bom dia! Obrigado, Professor @andrebaltieri. Estou usando a versão 7.0.102. Mesmo assim, acho estanho, pq vendo as aulas vc sempre executa o comando em um novo console e logo as alterações são vistas... aqui não, sempre preciso debugar ou limpar para depois fazer um run e ver as alterações do código no console.

**Reply by andrebaltieri** [reply]
@marcelloalonso eu to no Windows neste momento e está intermitente mesmo :(
As vezes ele recarrega, as vezes eu preciso dar um `CTRL+R` para forçar o recarregamento.

Acredito que seja algo da última release do .NET 7... vamos aguardar estabilizar...
